nativelink-scheduler/src/default_scheduler_factory.rs

```diff
@@ -150,6 +151,40 @@ fn simple_scheduler_factory(
             );
             Ok((Some(action_scheduler), Some(worker_scheduler)))
         }
+        ExperimentalSimpleSchedulerBackend::Mongo(mongo_config) => {
+            let store = store_manager
+                .get_store(mongo_config.mongo_store.as_ref())
+                .err_tip(|| {
+                    format!(
+                        "'mongo_store': '{}' does not exist",
+                        mongo_config.mongo_store
+                    )
+                })?;
+            let task_change_notify = Arc::new(Notify::new());
+            let store = store
+                .into_inner()
+                .as_any_arc()
+                .downcast::<ExperimentalMongoStore>()
```

Comment:
This downcast is causing some problems and I'm trying to figure out. I'm confident that Mongo can present a more usable option for some customers.